extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/ConfigurationService.java

```diff
@@ -223,4 +231,51 @@ public int getMaxResults() throws GuacamoleException {
         );
     }
 
+    /**
+     * Returns whether or not LDAP aliases will be dereferenced,
+     * as configured with guacamole.properties.  The default
+     * behavior if not explicityly defined is to never 
+     * dereference them.
+     *
+     * @return
+     *     The behavior for handling dereferencing of aliases
+     *     as configured in guacamole.properties.
+     *
+     * @throws GuacamoleException
+     *     If guacamole.properties cannot be parsed.
+     */
+    public DereferenceAliases getDereferenceAliases() throws GuacamoleException {
+        return environment.getProperty(
+            LDAPGuacamoleProperties.LDAP_DEREFERENCE_ALIASES,
+            DereferenceAliases.NEVER
+        );
+
+    }
+
+    /**
+     * Returns a set of LDAPSearchConstraints to apply globally
+     * to all LDAP searches rather than having various instances
+     * dispersed throughout the code.  Currently contains the
```

Comment:
I would recommend, simply:

"Returns a set of LDAPSearchConstraints which should be applied to all LDAP searches."

That a function should be used "rather than [copying the internals of the function everywhere]" applies to all functions by definition, and goes without saying. There's no need to enumerate the bad practices we avoid by having functions.

That said ... if this function effectively replaces the `getDereferenceAliases()` and `getMaxResults()` functions, and it no longer makes sense to call those functions externally, perhaps those functions should be removed (or declared `private`)?